pkg/aws/gather_test.go

```diff
@@ -0,0 +1,63 @@
+package aws
+
+import (
+	"context"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	utils "github.com/red-hat-storage/ocs-osd-deployer/testutils"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+)
+
+var _ = Describe("AWS Data Gathering behavior", func() {
+	ctrl.SetLogger(zap.New(zap.UseDevMode(true)))
+	log := ctrl.Log.WithName("aws-data-gather-test")
+	ctx := context.Background()
+
+	configMap := corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      DataConfigMapName,
+			Namespace: testNamespace,
+		},
+	}
+
```

Comment:
- if possible, before this test can we have below two tests?
1. Mocker is reachable within the time period
2. Mocker is unreachable within time period and assert the flow stops

Reply:
@leelavg I can't really think of a simple way to do this, would it be alright if we deferred this to perhaps a later pr?

Reply:
sure.